builder/azure/arm/step_delete_additional_disks.go

```diff
@@ -67,19 +67,13 @@ func (s *StepDeleteAdditionalDisk) Run(ctx context.Context, state multistep.Stat
 		dataDisks = disks.([]string)
 	}
 	var isManagedDisk = state.Get(constants.ArmIsManagedImage).(bool)
-	var isExistingResourceGroup = state.Get(constants.ArmIsExistingResourceGroup).(bool)
 	var resourceGroupName = state.Get(constants.ArmResourceGroupName).(string)
 
 	if dataDisks == nil {
 		s.say(" -> No Additional Disks specified")
 		return multistep.ActionContinue
 	}
 
-	if isManagedDisk && !isExistingResourceGroup {
```

Comment:
@nywilken this check has been in the code since this logic was implemented back in 2018, I think we don't properly clean up additional disks for managed images ever as shown in: https://github.com/hashicorp/packer-plugin-azure/issues/215, I don't think we should keep these additional VM disks regardless of if we're in an existing resource group we're using or not for the data disks but I'm not 100% certain my logic, what do you think?

Reply:
OH Actually this step isn't even used!!! That's why the additional disks aren't deleted, we rely on the cleanup of builder/azure/arm/step_create_resource_group.go to delete it, so if we don't delete the resource group we don't delete the additional disks!